README.md

```diff
@@ -128,68 +110,55 @@ var app = new EmberApp({
-Out of the box, we only enable:
+### Helper usage
 
- - css
- - coffeescript
- - html/xml
- - json
- - javascript
- - markdown
- - handlebars
- - htmlbars
+After you have defined your snippets, you can use the `get-code-snippet` helper to get the snippet data
+for rendering: `{{get-code-snippet "my-nice-example.js"}}`. The returned value will be a JavaScript object with the
+following properties:
 
-If you want a different set, you can:
+* `source`: the source code extracted from the given snippet
+* `language`: the snippets language, following the naming conventions of the popular `highlight.js` library, e.g. `htmlbars` for Ember templates
+* `extension`: the file extension of the file containing the given snippet
 
-1. Tell ember-code-snippet not to include highlight.js automatically for you. Also, include an array of file extensions corresponding to the languages you want to use. If ```snippetExtensions``` is not defined, the file extensions corresponding to the default list of supported languages will be used.
+By default, the helper will try to unindent the code block by
+removing whitespace characters from the start of each line until the
+code bumps up against the edge. You can disable this with:
 
-```js
-  // in ember-cli-build.js
-  var app = new EmberApp(defaults, {
-    includeHighlightJS: false,
-    snippetExtensions: ['js','java','php']
-  });
+```hbs
+{{get-code-snippet "my-nice-example.js" unindent=false}}
 ```
 
-2. Go to https://highlightjs.org/download/ and generate your own build of highlight.js using the languages you want.
+The following example will render a `<pre>` tag with a given code snippet:
 
-3. Place the resulting highlight.pack.js in your `vendor` directory.
-
-4. Import it directly from your ember-cli-build.js file:
-
-```js
-app.import('vendor/highlight.pack.js', {
-  using: [ { transformation: 'amd', as: 'highlight.js' } ]
-});
+```hbs
+{{#let (get-code-snippet "static.hbs") as |snippet|}}
+<pre class={{snippet.language}}>{{snippet.source}}</pre>
+{{/let}}
 ```
 
-# Theming Support
-
-We include a basic syntax-highlighting theme by default, but highlight.js has 79 different themes to choose from and it's possible to make your own just by writing a stylesheet.
-
-To use a different theme:
+### Syntax Highlighting
 
-1. Tell ember-code-snippet not to include its own theme:
+This addon does not provide any syntax highlighting capabilities itself, but instead it is designed with composability 
+in mind, so you can add highlighting capabilities with any highlighting library on top of the snippet extraction 
+primitives of this addon. The following is an example of rendering a code snippet using code highlighting provided by the 
+[ember-prism](https://github.com/shipshapecode/ember-prism) addon:
 
-```js
-  // in ember-cli-build.js
-  var app = new EmberApp(defaults, {
-    includeHighlightStyle: false
-  });
+```hbs
+{{#code-block language="handlebars"}}{{get (get-code-snippet "demo.hbs") "source"}}{{/code-block}}
```

Comment:
Can we use an example like this?

```hbs
{{#let (get-code-snippet "demo.hbs") as |snippet|}}
  {{#code-block language=snippet.language}}
    {{snippet.source]}
  {{/code-block}}
{{/let}}
```

It would show why we decided to return a POJO with the language too. This snippet can more easily generalize to a reusable component that accepts the snippet name as an argument:

```hbs
{{!-- templates/components/code-snippet.hbs --}}
{{#let (get-code-snippet @name) as |snippet|}}
  {{#code-block language=snippet.language}}
    {{snippet.source]}
  {{/code-block}}
{{/let}}
```

Which you would use like:

```hbs
<CodeSnippet @name="demo.hbs" />
```

Maybe we should even suggest that pattern in the README. I think most apps that will have many snippets should make a component like this that puts together their chosen syntax highlighter and get-code-snippet.

Reply:
Oh, and also we should explain this is the upgrade path for people who were using the current version of the addon. If they just want to keep the behavior they had, they should make their own `<CodeSnippet />` component to replace the one we removed.

Reply:
Makes sense, updated according to your suggestion!

> It would show why we decided to return a POJO with the language too

There is one potential problem though: that example would actually not work properly with prism.js and hbs templates, as our helper returns `'htmlbars'` as the language, while prism expects `'handlebars'`. But as that the language value is not standardized across highlighters, there is not so much we can do about it!?

Reply:
If prism is going to be our recommended syntax highlighter, I'm fine with changing our helper to return `handlebars`.

Reply:
I just realized I committed my recent Readme updates to the wrong branch. Fixed! 🤦‍♂️

> If prism is going to be our recommended syntax highlighter, I'm fine with changing our helper to return handlebars.

Ok, agree. Updated that, including a test + docs.